gateway-service/src/main/java/org/zowe/apiml/gateway/filters/post/PostStoreLoadBalancerCacheFilter.java

```diff
@@ -41,29 +41,16 @@ public int filterOrder() {
 
     @Override
     public boolean shouldFilter() {
-        return true;
+        Optional<InstanceInfo> instance = RequestContextUtils.getInstanceInfo();
+        return instance.isPresent() && metadataExists(instance.get()) && lbTypeExistsAuthenticationExistsInstanceIdExists(instance.get());
```

Comment:
It is tough for me to visually parse this, I get mixed up with what brackets are paired up. Would help to put each chained expression on a different line.